source/PlayerInfo.cpp

```diff
@@ -683,78 +683,43 @@ const Date &PlayerInfo::GetDate() const
 	return date;
 }
 
-
-
-// Set the date to the next day, and perform all daily actions.
-void PlayerInfo::IncrementDate()
+// Set the date, and perform all daily actions the given number of times.
+// Offers limited support for negative values.
+void PlayerInfo::AdvanceDate(int amount)
 {
-	++date;
-
-	// Check if any special events should happen today.
-	auto it = gameEvents.begin();
-	while(it != gameEvents.end())
+	if(!amount)
+		return;
+	else if(amount > 0)
 	{
-		if(date < it->GetDate())
-			++it;
-		else
+		while(amount--)
 		{
-			it->Apply(*this);
-			it = gameEvents.erase(it);
-		}
-	}
+			++date;
 
-	// Check if any missions have failed because of deadlines and
-	// do any daily mission actions for those that have not failed.
-	for(Mission &mission : missions)
-	{
-		if(mission.CheckDeadline(date) && mission.IsVisible())
-			Messages::Add("You failed to meet the deadline for the mission \"" + mission.Name() + "\".",
-				Messages::Importance::Highest);
-		if(!mission.IsFailed())
-			mission.Do(Mission::DAILY, *this);
-	}
+			// Check if any special events should happen today.
+			auto it = gameEvents.begin();
+			while(it != gameEvents.end() && date >= it->GetDate())
+			{
+				GameEvent event = *it;
+				event.Apply(*this);
```

Comment:
```suggestion
				(*it).Apply(*this);
```
sorry, I am irritated, wouldn't this be the same

Reply:
oh damn
I was trying to figure out why my code didn't work so I split it up
it was missing the parenthesis

Reply:
The existing code seems to copy the event pointed to by "it" and then apply that, which is not strictly the same as applying the event pointed to be "it".

Not sure it should be copying the event but they aren't quite the same.

In the case of using "it" directly, though, use `->`.

Reply:
the original code just uses ->, so I guess that would be the right thing to do?

Reply:
Okay, your version didn't work, it has some issues with `const`.

> error: passing ‘const GameEvent’ as ‘this’ argument discards qualifiers [-fpermissive]

I'll try the other one as well, but I have a feeling I already tried that.

Reply:
Yeah, that also doesn't work.